compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -492,18 +492,9 @@ class PostTyper extends MacroTransform with InfoTransformer { thisPhase =>
           )
         case Block(_, Closure(_, _, tpt)) if ExpandSAMs.needsWrapperClass(tpt.tpe) =>
           superAcc.withInvalidCurrentClass(super.transform(tree))
-        case _: Quote =>
+        case _: Quote | _: QuotePattern =>
           ctx.compilationUnit.needsStaging = true
           super.transform(tree)
-        case _: QuotePattern =>
-          if !ctx.reporter.errorsReported then
-            Checking.checkAppliedTypesIn(TypeTree(tree.tpe).withSpan(tree.span))
-          ctx.compilationUnit.needsStaging = true
-          super.transform(tree)
-        case tree: SplicePattern =>
-          if !ctx.reporter.errorsReported then
-            Checking.checkAppliedTypesIn(TypeTree(tree.tpe).withSpan(tree.span))
```

Comment:
This was a workaround that we added to check the types of the splices because we removed them in `QuotesAndSplices`. Now those types are checked directly on their ASTs.